### PR TITLE
chore(EMS-1969): Additional application fields, prevent submission if submissionCount is not 0

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR adds 2 new fields to applications - `submissionCount` and `dealType`. This is so that we are aligned with other products and in the future, EXIP applications can be easily queried/mappeed.

## Changes
- Update schema to include new fields.
- When an application is created, add a default submission count and deal type.
- When an application is submitted:
  - Check that the `submissionCount` is 0 - otherwise reject. (We do not allow an application to be submitted more than once).
  - Increment the `submissionCount`.
- Update the UI's `canSubmitApplication` function to check the `submissionCount`.
- Update types and mocks.

### CODE REVIEW

File: database/exip.sql [hunk 1]
```diff
@@ -134,6 +135,7 @@ CREATE TABLE `Application` (
 	`sectionReview` varchar(191) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
 	`owner` varchar(191) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
 	`version` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '1',
+	`dealType` varchar(191) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT 'EXIP',
```
Comment (abhi-markan) [inline database/exip.sql:138]: Do we need varchar(191) here, we can just set it to `VARCHAT(4)`?

File: src/api/custom-resolvers/mutations/submit-application/index.ts [hunk 2]
```diff
@@ -27,21 +27,24 @@ const submitApplication = async (root: any, variables: SubmitApplicationVariable
...
+      const isInProgress = status === APPLICATION.STATUS.IN_PROGRESS;
 
       const now = new Date();
 
       // check the current date vs submission deadline
-      const validSubmissionDate = isAfter(new Date(application.submissionDeadline), now);
+      const validSubmissionDate = isAfter(new Date(submissionDeadline), now);
 
-      const canSubmit = isInProgress && validSubmissionDate;
+      const canSubmit = isInProgress && validSubmissionDate && submissionCount === 0;
```
Comment (abhi-markan) [inline src/api/custom-resolvers/mutations/submit-application/index.ts:39]: Might be ideal to create a variable as below

```javascript
const firstSubmission = submissionCount === 0;
const canSubmit = isInProgress && validSubmissionDate && firstSubmission;
```

File: src/ui/server/helpers/can-submit-application/index.ts [hunk 3]
```diff
@@ -6,25 +6,33 @@ import { Application } from '../../../types';
...
   const isInProgress = status === APPLICATION.STATUS.IN_PROGRESS;
 
-  // check the current date vs submission deadline
+  // check that the current date is before the submission deadline
   const now = new Date();
   const validSubmissionDate = isAfter(new Date(submissionDeadline), now);
 
-  if (isComplete && isInProgress && validSubmissionDate) {
+  // check that it has not been submitted before
+  const zeroSubmissionCount = submissionCount === 0;
```
Comment (abhi-markan) [inline src/ui/server/helpers/can-submit-application/index.ts:33]: `firstSubmission` perhaps?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Nice work, few minor comments.

File: src/api/custom-resolvers/mutations/submit-application/index.ts (hunk 2 above)
Comment (ttbarnes) [reply src/api/custom-resolvers/mutations/submit-application/index.ts:39]: Good shout, have changed and using `isFirstSubmission` variable.

File: src/ui/server/helpers/can-submit-application/index.ts (hunk 3 above)
Comment (ttbarnes) [reply src/ui/server/helpers/can-submit-application/index.ts:33]: Renamed to `isFirstSubmission`

**Review by Zainzzkk — APPROVED** [review]

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**